File: Chorme_extension/contentScript/All-in-JPEG/content-info/AudioContentInfo.js
import ContentAttribute from "../content/contentType.js"
export default class AudioContentInfo {
    constructor(audioContent, audioDataStartOffset) {
        this.contentInfoSize = 0;
        this.dataStartOffset = 0;
        this.datasize = 0;
        this.attribute = 0;
        this.FIELD_SIZE = 4;
        this.XOA_MARKER_SIZE = 2;

        this.init(audioContent, audioDataStartOffset);
    }

    init(audioContent, audioDataStartOffset) {
        this.contentInfoSize = this.FIELD_SIZE * 4;
        this.dataStartOffset = audioDataStartOffset;

        if (audioContent.aiAudio !== null) {
            this.attribute = audioContent.aiAudio.attribute;
            this.datasize = this.XOA_MARKER_SIZE + audioContent.aiAudio._audioByteArray.length;
        } else {
            this.attribute = ContentAttribute.basic.code;
            this.datasize = 0;
        }
    }

    convertBinaryData() {
        const buffer = new ArrayBuffer(this.contentInfoSize);
        const dataView = new DataView(buffer);

        dataView.setInt32(0, this.contentInfoSize);
        dataView.setInt32(4, this.dataStartOffset);
        dataView.setInt32(8, this.attribute);
        dataView.setInt32(12, this.datasize);

        return new Uint8Array(buffer);
    }

    getLength() {
        return this.contentInfoSize;
    }
}

File: OnePIC/app/src/main/java/com/goldenratio/onepic/AllinJPEGModule/ContentInfo/AudioContentInfo.kt
package com.goldenratio.camerax.PictureModule.Info

import com.goldenratio.onepic.AllinJPEGModule.Content.AudioContent
import com.goldenratio.onepic.AllinJPEGModule.Content.ContentAttribute
import java.nio.ByteBuffer

class AudioContentInfo(audioContent: AudioContent, audioDataStartOffset : Int) {
    var contentInfoSize : Int = 0
    var dataStartOffset : Int = 0
    var datasize : Int = 0
    var attribute : Int

    companion object{
        val FIELD_SIZE = 4
        val XOA_MARKER_SIZE = 2
    }

    init{
        // contentInfo size
        contentInfoSize = FIELD_SIZE * 4
        dataStartOffset = audioDataStartOffset
        // size
        if(audioContent.audio != null){
            attribute = audioContent.audio!!.attribute.code
            datasize = XOA_MARKER_SIZE + audioContent.audio!!._audioByteArray!!.size
        }else {
            attribute = ContentAttribute.basic.code
            datasize = 0
        }
    }
    fun converBinaryData(): ByteArray {
        val buffer: ByteBuffer = ByteBuffer.allocate(contentInfoSize)
        //Audio Content
        buffer.putInt(contentInfoSize)
        buffer.putInt(dataStartOffset)
        buffer.putInt(attribute)
        buffer.putInt(datasize)
        return buffer.array()
    }
    fun getLength() : Int {
        return contentInfoSize
    }
}